src/Test/Monad/Reader.hs

```diff
@@ -10,12 +10,20 @@ import Test.QuickCheck.HigherOrder (Equation(..))
 
 -- * Primary laws
 
+-- | 'ask' is idempotent.
+-- @
+-- 'ask' '>>' 'ask' = 'ask'
+-- @
 ask_ask
   :: forall m r
   .  MonadReader r m
   => Equation (m r)
 ask_ask = (ask >> ask) :=: ask @r @m
 
+-- | 'local' can be interchanged with fmap.
```

Comment:
Not quite. This law works because `ask` is the second argument.

To modify the environment then asking for it is the same as asking for the environment then modifying it.